src/main/java/graphql/schema/idl/SchemaPrinter.java

```diff
@@ -306,12 +354,33 @@ private boolean isIntrospectionType(GraphQLType type) {
                     .collect(toList());
             for (GraphQLEnumValueDefinition enumValueDefinition : values) {
                 printComments(out, enumValueDefinition, "  ");
-                out.format("  %s%s\n", enumValueDefinition.getName(), directivesString(GraphQLEnumValueDefinition.class, enumValueDefinition.getDirectives()));
+                List<GraphQLDirective> enumValueDirectives = enumValueDefinition.getDirectives();
+                if (enumValueDefinition.isDeprecated()) {
+                    enumValueDirectives = addDeprecatedDirectiveIfNeeded(enumValueDirectives);
```

Comment:
I see you have this private method to help adding the deprecated directive to the list if it does not exist. Is there a reason why this is a list instead of a set in which case this private method wouldn't be needed.